### PR TITLE
deps: update Bazel to v7

### PR DESCRIPTION
**Pull request by malt3**
### Context

This updates Bazel to a new major version.
I also updated our nix flake and mkosi with it.

### Proposed change(s)

- Update Bazel to v7
- Fix issues with bazel_7 on NixOS
- Update mkosi

### Checklist

- [x] Run the E2E tests that are relevant to this PR's changes
- [x] Update [docs](https://github.com/edgelesssys/constellation/tree/main/docs)
- [x] Add labels (e.g., for changelog category)
- [x] Is PR title adequate for changelog?
- [x] Link to Milestone
- [x] Build image for testing: `ref/deps-bazel-bazel-7/stream/debug/v2.16.0-pre.0.20240216171549-80c09f5b3149`
- [x] Reproducible builds test: https://github.com/edgelesssys/constellation/actions/runs/7932992321
- [x] [e2e for good measure](https://github.com/edgelesssys/constellation/actions/runs/7934078801)

### CODE REVIEW

File: .bazelrc [hunk 1]
```diff
@@ -1,5 +1,8 @@
+# Sadly, some Bazel rules we depend on have no support for bzlmod yet
```
Comment (burgerdev) [inline .bazelrc:1]: Can we document them so that it's easier to check for support later?

File: bazel/bazelrc/correctness.bazelrc [hunk 2]
```diff
@@ -44,3 +44,16 @@ query --experimental_allow_tags_propagation
...
+# because it contains a BUILD file. See https://github.com/bazelbuild/bazel/issues/8195.
+# Docs: https://bazel.build/reference/command-line-reference#flag--incompatible_disallow_empty_glob
+common --incompatible_disallow_empty_glob
+
+# Always download coverage files for tests from the remote cache. By default, coverage files are not
+# downloaded on test result cahce hits when --remote_download_minimal is enabled, making it impossible
+# to generate a full coverage report.
+# Docs: https://bazel.build/reference/command-line-reference#flag--experimental_fetch_all_coverage_outputs
+# detching remote cache results
+test --experimental_fetch_all_coverage_outputs
```
Comment (burgerdev) [inline bazel/bazelrc/correctness.bazelrc:59]: Is this a universally useful thing, or do we only need that when actually generating coverage reports?

File: bazel/bazelrc/correctness.bazelrc [hunk 3]
```diff
@@ -44,3 +44,16 @@ query --experimental_allow_tags_propagation
...
 build --incompatible_default_to_explicit_init_py
+
+# Set default value of `allow_empty` to `False` in `glob()`. This prevents a common mistake when
+# attempting to use `glob()` to match files in a subdirectory that is opaque to the current package
+# because it contains a BUILD file. See https://github.com/bazelbuild/bazel/issues/8195.
+# Docs: https://bazel.build/reference/command-line-reference#flag--incompatible_disallow_empty_glob
+common --incompatible_disallow_empty_glob
+
+# Always download coverage files for tests from the remote cache. By default, coverage files are not
+# downloaded on test result cahce hits when --remote_download_minimal is enabled, making it impossible
```
Comment (burgerdev) [inline bazel/bazelrc/correctness.bazelrc:55]:
```suggestion
# downloaded on test result cache hits when --remote_download_minimal is enabled, making it impossible
```

File: bazel/container/Containerfile
Comment (burgerdev) [inline bazel/container/Containerfile:1]: This is still referenced from `CODEOWNERS` and `.github/workflows/build-bazel-container.yml`.

File: bazel/bazelrc/correctness.bazelrc (hunk 2 above)
Comment (malt3) [reply bazel/bazelrc/correctness.bazelrc:59]: It is generally useful and copied from https://github.com/aspect-build/bazel-lib/pull/564

File: bazel/bazelrc/correctness.bazelrc (hunk 3 above)
Comment (malt3) [reply bazel/bazelrc/correctness.bazelrc:55]: Oops. Copied the typo from aspect as well

File: .bazelrc (hunk 1 above)
Comment (malt3) [reply .bazelrc:1]: I added a link to the upstream rules_nixpkgs issue. Also mentioned that there may be more we are not seeing yet.

**Review by daniel-weisse — APPROVED** [review]
changes to `/internal/attestation` look good to me

**Review by burgerdev — APPROVED** [review]

**Pull request merged**